ArmTemplates/app-gateway-v2.md

```diff
@@ -0,0 +1,239 @@
+    "port": 443,
+    "protocol": "Https",
+    "hostnameFromBackendAddress": true
+}
+```
+
+routingRules: (required) array of object
+
+A list of routing rules describing which paths should be routed to which backend pool.
+If the path does not match any rule provided it will go to the default backend pool,
+which is the first backend specified (see above).
+
+Each backend is specified by an object consisting of
+
+* name: the name the backend pool resource
+* backendPool: the name of the backend to route to (as specified in a previous parameter, see above)
+* backendHttp: the name of the backend http settings to use (as specified in the previous parameter, see above)
+* paths: an array of paths, usually wildcarded, to route to the backend
+
+An example of a valid object
+
+```json
+{
+    "name": "routingRule",
+    "backendPool": "backendName",
+    "backendHttp": "httpSettingName",
+    "paths": [ "/myapp/*" ]
+}
+```
+
+customProbes: (optional) array of objects
+
+Create probes for use in backendHttpSettings.
+See https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2018-11-01/applicationgateways#ApplicationGatewayProbe
+for options.
+
+An example of a valid object array
+
+```json
+[
+    {
+        "name": "myProbe",
+        "type": "Microsoft.Network/applicationGateways/probes",
+        "properties": {
+            "protocol": "Https",
+            "path": "/",
+            "interval": 60,
+            "timeout": 30,
+            "unhealthyThreshold": 3,
+            "pickHostNameFromBackendHttpSettings": true,
+            "minServers": 0,
+            "match": {
+                "statusCodes": [
+                    "200-399"
+                ]
+            }
+        }
+    }
+]
+```
+
+customErrorPages: (optional) array of objects
+
+Create probes for use in backendHttpSettings.
+See https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2018-11-01/applicationgateways#ApplicationGatewayCustomError
+for options.
+
+An example of a valid object array
+
+```json
+[
+    {
+        "statusCode": "HttpStatus502",
+        "customErrorPageUrl": "https://my.domain/502.html"
+    }
+]
+```
+
+rewriteRules: (optional) array of object
+
+A list of rewrite rules which will be applied to all URL paths.
+If not specified, no rewrite rules will be specified.
+This is only valid with v2 tiers.
+
+Each rewrite rule is specified by an object consisting of
+
+* name: the name the rewrite rule set
+* ruleSequence: an integer specifying the order to run the rules (lowest to highest) - defaults to 100 if not specified
+* conditions: an array of objects specifying the conditions that need to be met for the rule to be applied - rules are run unconditionally if not specified
+* actionSet: an array of objects specifying actionSet of the rule
+
+An example of a valid object. Only name and actionSet are required.
+
+```json
+{
+    "name": "rewriteRule",
+    "ruleSequence": 100,
+    "conditions": { ... }, 

```

Comment:
these are objects but documentation above says they should be arrays

Reply:
Fixed and included a full working example